packages/apollo-server-fastify/src/__tests__/ApolloServer.test.ts

```diff
@@ -0,0 +1,835 @@
+      );
+
+      const apolloFetch = createApolloFetch({ uri }).useAfter(
+        (response, next) => {
+          expect(
+            response.response.headers.get('access-control-allow-origin'),
+          ).toEqual('apollographql.com');
+          next();
+        },
+      );
+      await apolloFetch({ query: '{hello}' });
+    });
+
+    describe('healthchecks', () => {
+      afterEach(async () => {
+        await server.stop();
+      });
+
+      it('creates a healthcheck endpoint', async () => {
+        const { port } = await createServer({
+          typeDefs,
+          resolvers,
+        });
+
+        return new Promise((resolve, reject) => {
+          request(
+            {
+              url: `http://localhost:${port}/.well-known/apollo/server-health`,
+              method: 'GET',
+            },
+            (error, response, body) => {
+              if (error) {
+                reject(error);
+              } else {
+                expect(body).toEqual(JSON.stringify({ status: 'pass' }));
+                expect(response.statusCode).toEqual(200);
+                resolve();
+              }
+            },
+          );
+        });
+      });
+
+      it('provides a callback for the healthcheck', async () => {
+        const { port } = await createServer(
+          {
+            typeDefs,
+            resolvers,
+          },
+          {
+            onHealthCheck: async () => {
+              throw Error("can't connect to DB");
+            },
+          },
+        );
+
+        return new Promise((resolve, reject) => {
+          request(
+            {
+              url: `http://localhost:${port}/.well-known/apollo/server-health`,
+              method: 'GET',
+            },
+            (error, response, body) => {
+              if (error) {
+                reject(error);
+              } else {
+                expect(body).toEqual(JSON.stringify({ status: 'fail' }));
+                expect(response.statusCode).toEqual(503);
+                resolve();
+              }
+            },
+          );
+        });
+      });
+
+      it('can disable the healthCheck', async () => {
+        const { port } = await createServer(
+          {
+            typeDefs,
+            resolvers,
+          },
+          {
+            disableHealthCheck: true,
+          },
+        );
+
+        return new Promise((resolve, reject) => {
+          request(
+            {
+              url: `http://localhost:${port}/.well-known/apollo/server-health`,
+              method: 'GET',
+            },
+            (error, response) => {
+              if (error) {
+                reject(error);
+              } else {
+                expect(response.statusCode).toEqual(404);
+                resolve();
+              }
+            },
+          );
+        });
+      });
+    });
+    // NODE: Intentionally skip file upload tests on Node.js 10 or higher.
+    (atLeastMajorNodeVersion(10) ? describe.skip : describe)(
```

Comment:
Some slight tweaks to note in order to avoid conflicting with #2054.  

(Presumably we'll just want to use `NODE_MAJOR_VERSION !== 6 ? describe : describe.skip`.)